### ISSUE TITLE
[Feature Request] Use `fun` keyword for functions

### ISSUE THREAD

**Issue by redradist** [modularml/mojo#453]
### Review Mojo's priorities

- [X] I have read the [roadmap and priorities](https://docs.modular.com/mojo/roadmap.html#overall-priorities) and I believe this request falls within the priorities.

### What is your request?

```shell
Currently static function keyword was inspired by Rust, but `fn` is strange keyword even in Rust, because `struct` did not remove any intermediate characters, but for function letter `u` was remove, not only it make design inconsistent, but it also hard to read it if you a newbie ...

For Python context, switching from `fn` -> `fun` will improve aligment with currect Python functions.

def login(name):
    pass

fn login(name, token): # login name of function is not at the same position like previous function
    pass

fun login(name, token): # Much better than previous example for alignment
    pass
```

### What is your motivation for this change?

```shell
Make more consistent syntax and better formatting
```

**Comment by gryznar** [comment]
For me a "fn" is a better abbreviation for function than "fun". "fun" in english is a separate word, while "fn" is not 🙃

**Comment by immmor** [comment]
I prefer 'fun', which will bring more fun(^_^) during the coding experience~

**Comment by redradist** [comment]
@immmor 
> I prefer 'fun', which will bring more fun(^_^) during the coding experience~

Yes, it is also one of the reason that I like this keyword )

**Comment by tereigo** [comment]
Makes sense to me!

**Comment by cooperliu101** [comment]
How do you pronounce 'fn'? 
fun? f n? function? fin?

**Comment by UnionChips** [comment]
You have unused fun in your present... I like it 😊🎁

**Comment by redradist** [comment]
> How do you pronounce 'fn'? fun? f n? function? fin?

@cooperliu101 `fun`, just like `def` - def_inition, `fun` - fun_ction
Also `fun` name will give good alignment of `fun` signature with `def` signature

**Comment by Marcus-Peterson** [comment]
Honestly kind of ridiculous (no pun intended) mojo is supposed to be an extension of python. And one of the core features of python is: simplicity. fn encapsulates that. 

Two letters, nothing more, nothing less

**Comment by WIZARDELF** [comment]
Viewing from aesthetics, `fn` is ugly. *mojo* aims to extend *python*, so why do we simply extend `def` with `immutable def` or similar mechanism. Personally, I prefer `fun`, or a new thing like `funk`, better than `fn`, smelling unhealthy.

**Comment by Marcus-Peterson** [comment]
Yeah, your point may be valid. But in the end, these are small details that do not impact overall code quality, compilation etc. 
The fact that I, you and everybody else in this comment section is worried wether we should add another letter to how we declare functions is ridiculous and is misplaced focus.

**Comment by gryznar** [comment]
"fun" is much weirder than "fn"

**Comment by ParadaCarleton** [comment]
> Honestly kind of ridiculous (no pun intended) mojo is supposed to be an extension of python. And one of the core features of python is: simplicity. fn encapsulates that.
> 
> Two letters, nothing more, nothing less

Pythn nvr, _vr_, ss bjd txt fr kywrds r rsrvd wrds. Pythn lwys nclds vwls, bcs nglsh s nt n bjd lngg. 

<details> Python never, _ever_, uses abjad text for keywords or reserved words. Python always includes vowels, because English is not an abjad language.</details>

Personally, I don't care that much about consistency with Python, so I don't think this matters that much. But if you do want to stick close to Python, `fn` isn't the kind of syntax Python would use. Python uses full English words (sometimes truncated), but it doesn't drop vowels, because dropping vowels some of the time makes it hard to predict what functions have missing vowels.

**Comment by redradist** [comment]
> > Honestly kind of ridiculous (no pun intended) mojo is supposed to be an extension of python. And one of the core features of python is: simplicity. fn encapsulates that.
> > Two letters, nothing more, nothing less
> 
> Pythn dsnt s bjd txt fr kywrds r rsrvd wrds. Pythn lwys nclds vwls, bcs nglsh s nt n bjd lngg.
> Python doesn't use abjad text for keywords or reserved words. Python always includes vowels, because English is not an abjad language.
> 
> Personally, I don't care that much about consistency with Python, so I don't think this matters that much. But if you do want to stick close to Python, `fn` isn't the kind of syntax Python would use. Python uses full English words (sometimes truncated), but it doesn't drop vowels, because dropping vowels some of the time makes it hard to predict what functions have missing vowels.

Consistent language is made by small parts, each peace makes language consistent to itself.
For example, @lattner in podcast of Lex Freedman said that they choice the Python indentation mechanism, but does it matter if basic sytax do not allow proper indentation ?
I mean changing `fn` -> `fun` is such simple, but will 'cause much simpler change from `def` -> `fun` and `fun` -> `fun` backward, small feature that improve consistency and it improve ability easier change from one concept to another

**Comment by josiahls** [comment]
@JoeLoser  I think the bike shedding on this is done isn't it?
